src/base/os.lua

```diff
@@ -387,10 +391,17 @@
 				while os.matchnext(m) do
 				if not (slashpos and os.matchisfile(m)) then
 					local matchpath = path.join(before, matchpath, os.matchname(m))
+					
 					if after then
-						results = table.join(results, os.match(path.join(matchpath, after)))
+						results = table.join(results, os.match(path.join(matchpath, after), matchType))
 					else
-						table.insert(results, matchpath)
+						if matchType == "file" and os.matchisfile(m) then
```

Comment:
typo? for below too
```suggestion
						if matchType == "file" and os.matchisfile(matchpath) then
```

Reply:
We are in `if not (slashpos and os.matchisfile(m)) then` block...  so `os.matchisfile(m)` is always false...

Reply:
That's not how `not (A and B)` works, either `A` or `B` can be false.

Reply:
Oops, indeed, misread.